lib/setup/SetupClass.php

```diff
@@ -724,10 +724,16 @@ public function drop()
             if (!$bFound) array_push($aDropTables, $sTable);
         }
         foreach ($aDropTables as $sDrop) {
-            if ($this->bVerbose) echo "Dropping table $sDrop\n";
-            $this->oDB->exec("DROP TABLE $sDrop CASCADE");
-            // ignore warnings/errors as they might be caused by a table having
-            // been deleted already by CASCADE
+            if ($this->oDB->table_exists($sDrop)) {
+                if ($this->bVerbose) echo "Dropping table $sDrop\n";
+                try {
+                    $this->oDB->exec("DROP TABLE $sDrop CASCADE");
+                } catch (\Nominatim\DatabaseError $e) {
+                    // ignore warnings/errors as they might be caused by a table having
+                    // been deleted already by CASCADE
+                    if ($this->bVerbose) warn($e->getMessage());
+                }
```

Comment:
As you are doing it right now and check if the table exists before dropping. Shouldn't there also be no errors now?

Reply:
Why not

```sql
DROP TABLE IF EXISTS $sDrop CASCADE
```

And let Postgres handle this?

Reply:
Yes, seems, easier